File: tests/golang/bitset_test.go
package tests

import (
	"testing"

	_type "github.com/Alber-Blanc/messgen/build-golang-test/msgs/mynamespace/types"
)

func TestBitset(t *testing.T) {
	bitset := _type.SimpleBitset(0)

	bitset.Set(_type.SimpleBitset_One)
	bitset.Set(_type.SimpleBitset_Error)

	if bitset != _type.SimpleBitset(0b101) {
		t.Errorf("Expected bitset to be 0b101, got 0b%b", bitset)
	}

	if !bitset.Has(_type.SimpleBitset_One) {
		t.Errorf("Expected bitset to have 'One' bit set")
	}

	if bitset.Has(_type.SimpleBitset_Two) {
		t.Errorf("Expected bitset to not have 'Two' bit set")
	}

	if !bitset.Has(_type.SimpleBitset_Error) {
		t.Errorf("Expected bitset to have 'Error' bit set")
	}

	expectedString := "{one | error}"

	if bitset.String() != expectedString {
		t.Errorf("Expected bitset string to be %s, got %s", expectedString, bitset.String())
	}

	bitset.Clear(_type.SimpleBitset_One)
	if bitset != _type.SimpleBitset(0b100) {
		t.Errorf("Expected bitset to be 0b100 after clearing 'One', got 0b%b", bitset)
	}

	bitset.Clear(_type.SimpleBitset_Error)
	if bitset != _type.SimpleBitset(0b000) {
		t.Errorf("Expected bitset to be 0b000 after clearing 'Error', got 0b%b", bitset)
	}
}

File: tests/dart/test/bitset_test.dart
import 'package:test/test.dart';
import '../../../build-dart-test/msgs/mynamespace/types/simple_bitset_gen.dart';

void main() {
  group('Bitset Tests', () {
    test('Bitset set and has operations', () {
      final bitset = SimpleBitset(0);

      bitset.set(SimpleBitset.one);
      bitset.set(SimpleBitset.error);

      expect(bitset.value, int.parse('101', radix: 2));
      expect(bitset.has(SimpleBitset.one), isTrue);
      expect(bitset.has(SimpleBitset.two), isFalse);
      expect(bitset.has(SimpleBitset.error), isTrue);
    });

    test('Bitset clear operation', () {
      final bitset = SimpleBitset(int.parse('111', radix: 2));

      expect(bitset.has(SimpleBitset.one), isTrue);
      expect(bitset.has(SimpleBitset.two), isTrue);
      expect(bitset.has(SimpleBitset.error), isTrue);

      bitset.clear(SimpleBitset.two);

      expect(bitset.value, int.parse('101', radix: 2));
      expect(bitset.has(SimpleBitset.one), isTrue);
      expect(bitset.has(SimpleBitset.two), isFalse);
      expect(bitset.has(SimpleBitset.error), isTrue);
    });

    test('Bitset toString', () {
      final bitset = SimpleBitset(0);
      bitset.set(SimpleBitset.one);
      bitset.set(SimpleBitset.error);

      final str = bitset.toString();
      expect(str, equals('{one | error}'));
    });

    test('Empty bitset toString', () {
      final bitset = SimpleBitset(0);
      expect(bitset.toString(), equals('{}'));
    });
  });
}
